sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -106,21 +106,31 @@ private[hive] class HiveClientImpl(
 
     // Set up kerberos credentials for UserGroupInformation.loginUser within
     // current class loader
-    // Instead of using the spark conf of the current spark context, a new
-    // instance of SparkConf is needed for the original value of spark.yarn.keytab
-    // and spark.yarn.principal set in SparkSubmit, as yarn.Client resets the
-    // keytab configuration for the link name in distributed cache
```

Comment:
this seems odd, the comment seems to say that it was creating new instance of SparkConf but I don't see that in the code you removed so I guess this comment was just wrong... 

It looks like this must have gotten broke when we went to spark 2.x, original version actually was creating the new SparkConf:
https://github.com/apache/spark/pull/9272/files